### PR TITLE
Feature consume api library

### PR DESCRIPTION
**Pull request by AndreyWolox**
## Summary
- Add Dto models
- Add library service
- Add findByIsbn in controller and repository
- ADD mapper and rest beans
## Trello Card
https://trello.com/c/ONm5vCZl/22-integraci%C3%B3n-api-externa

### CODE REVIEW

File: src/main/java/wolox/training/controllers/BookController.java [hunk 1]
```diff
@@ -125,4 +132,24 @@ public Book updateBook(@RequestBody Book book, @PathVariable Long id) {
...
+     *
+     * @param isbn: Book isbn (String)
+     *
+     * @return the book by the isbn
+     * @exception ResponseStatusException if a book don´t exists in api library
+     */
+    @GetMapping("/isbn/{isbn}")
+    public Book findByIsbn(@PathVariable String isbn, HttpServletResponse response) {
+        response.setStatus(200);
+        return repository.findFirstByIsbn(isbn).orElseGet(() -> searchBookInLibrary(isbn, response));
```
Comment (Jimenarosello) [inline src/main/java/wolox/training/controllers/BookController.java:147]: Awesome!! :)

File: src/main/java/wolox/training/controllers/BookController.java [hunk 2]
```diff
@@ -125,4 +132,24 @@ public Book updateBook(@RequestBody Book book, @PathVariable Long id) {
...
+     * Find book by isbn
+     *
+     * @param isbn: Book isbn (String)
+     *
+     * @return the book by the isbn
+     * @exception ResponseStatusException if a book don´t exists in api library
+     */
+    @GetMapping("/isbn/{isbn}")
+    public Book findByIsbn(@PathVariable String isbn, HttpServletResponse response) {
+        response.setStatus(200);
```
Comment (Jimenarosello) [inline src/main/java/wolox/training/controllers/BookController.java:146]: I recommend using constants instead (check this out! https://docs.oracle.com/javaee/6/api/javax/servlet/http/HttpServletResponse.html) 😄

File: src/main/java/wolox/training/utils/BookConverter.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+import wolox.training.models.BookLibraryDto;
+import wolox.training.models.Publisher;
+
+public class BookConverter {
+
+    public static Book convertBookDtoToBookDao(BookLibraryDto bookLibraryDto) {
+        Book book = new Book();
+        book.setIsbn(bookLibraryDto.getIsbn());
+        book.setAuthor(bookLibraryDto.getAuthors().stream().map(Author::getName).collect(Collectors.joining(" - ")));
+        book.setPublisher(bookLibraryDto.getPublishers().stream().map(Publisher::getName).collect(Collectors.joining(" - ")));
```
Comment (Jimenarosello) [inline src/main/java/wolox/training/utils/BookConverter.java:15]: Nice!! :)

**Review by Jimenarosello — APPROVED** [review]
Awesome! 🚀

**Review by CarlosLugo22 — APPROVED** [review]
well done

**Pull request merged**